### ISSUE TITLE
process/link call increases exit message delivery time

### ISSUE THREAD

**Issue by sskorokhodov** [suprematic/otplike#12]
This test fails:

```clojure
(deftest exit-self-reason-is-process-exit-reason
  (let [done (async/chan)
        done1 (async/chan)
        pfn (proc-fn []
              (await-completion done1 50)
              (process/exit (process/self) :abnormal))
        pid (process/spawn pfn [] {})
        pfn1 (proc-fn[]
               (process/link pid)
               (async/close! done1)
               (is (= [:exit [pid :abnormal]] (<! (await-message 50)))
                   "process exit reason must be the one passed to exit call")
               (async/close! done))]
    (process/spawn pfn1 [] {:link-to pid :flags {:trap-exit true}})
    (await-completion done 100)))
```

`pfn1` receives `:timeout` instead of exit message.

And this test passes:

```clojure
(deftest ^:parallel exit-self-reason-is-process-exit-reason
  (let [done (async/chan)
        done1 (async/chan)
        pfn (proc-fn []
              (await-completion done1 50)
              (process/exit (process/self) :abnormal))
        pid (process/spawn pfn [] {})
        pfn1 (proc-fn[]
               ;(process/link pid) <<< this line is commented out
               (async/close! done1)
               (is (= [:exit [pid :abnormal]] (<! (await-message 50)))
                   "process exit reason must be the one passed to exit call")
               (async/close! done))]
    (process/spawn pfn1 [] {:link-to pid :flags {:trap-exit true}})
    (await-completion done 100)))
```

**Comment by sskorokhodov** [comment]
What happens:

1. `pfn1` releases `pfn` by closing `done1`.
2. `pfn`'s process exits.
3. `pfn1` tries to link `pfn`'s process. As long as `pfn`'s process already exited`, `process/link` dispatching blocks for timeout, preventing other control messages from being processed.
4. exit message is sent to `pfn1` as (delayed) result of `process/exit` call in `pfn`.
5. `pfn1`'s link fails by timeout (100 ms), letting exit message from previous item to be processed.

**Comment by sskorokhodov** [comment]
Fixed by 27c536af63df26d927f3cdf449bd61017cf80e6f
